pkg/controllers/cluster/cluster_controller.go

```diff
@@ -219,16 +230,21 @@ func (c *Controller) syncCluster(ctx context.Context, cluster *clusterv1alpha1.C
 	return c.ensureFinalizer(cluster)
 }
 
-func (c *Controller) removeCluster(cluster *clusterv1alpha1.Cluster) (controllerruntime.Result, error) {
-	err := c.removeExecutionSpace(cluster)
-	if err != nil {
-		klog.Errorf("Failed to remove execution space %v, err is %v", cluster.Name, err)
-		c.EventRecorder.Event(cluster, corev1.EventTypeWarning, fmt.Sprintf("Failed %s", clusterv1alpha1.EventReasonCreateExecutionSpaceFailed), err.Error())
+func (c *Controller) removeCluster(ctx context.Context, cluster *clusterv1alpha1.Cluster) (controllerruntime.Result, error) {
+	// add terminating taint before cluster is deleted
+	if err := utilhelper.UpdateClusterControllerTaint(ctx, c.Client, []*corev1.Taint{TerminatingTaintTemplate}, nil, cluster); err != nil {
+		c.EventRecorder.Event(cluster, corev1.EventTypeWarning, clusterv1alpha1.EventReasonRemoveTargetClusterFailed, err.Error())
+		klog.ErrorS(err, "Failed to update terminating taint", "cluster", cluster.Name)
```

Comment:
Here we assume there is no toleration for `cluster.karmada.io/terminating`, so the `taint manager` would finish the cleanup soon without waiting, am I right?

Reply:
That's right.